src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+        public bool Equals(DateOnly value) => _dayNumber == value._dayNumber;
+
+        /// <summary>
+        /// Returns a value indicating whether this instance is equal to a specified object.
+        /// </summary>
+        /// <param name="value">The object to compare to this instance.</param>
+        /// <returns>true if value is an instance of DateOnly and equals the value of this instance; otherwise, false.</returns>
+        public override bool Equals(object? value)
+        {
+            if (value is DateOnly)
+            {
+                return _dayNumber == ((DateOnly)value)._dayNumber;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns the hash code for this instance.
+        /// </summary>
+        /// <returns>A 32-bit signed integer hash code.</returns>
+        public override int GetHashCode() => _dayNumber;
+
+        // Only Allowed DateTimeStyles: AllowWhiteSpaces, AllowTrailingWhite, AllowLeadingWhite, and AllowInnerWhite
+
+        /// <summary>
+        /// Converts a memory span that contains string representation of a date to its DateOnly equivalent by using the conventions of the current culture.
+        /// </summary>
+        /// <param name="s">The memory span that contains the string to parse.</param>
+        /// <returns>An object that is equivalent to the date contained in s.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s) => Parse(s, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
+
+
+        private const ParseFlags ParseFlagsDateMask = ParseFlags.HaveHour | ParseFlags.HaveMinute | ParseFlags.HaveSecond | ParseFlags.HaveTime | ParseFlags.TimeZoneUsed |
+                                                      ParseFlags.TimeZoneUtc | ParseFlags.CaptureOffset | ParseFlags.UtcSortPattern;
+
+        /// <summary>
+        /// Converts a memory span that contains string representation of a date to its DateOnly equivalent by using culture-specific format information and a formatting style.
+        /// </summary>
+        /// <param name="s">The memory span that contains the string to parse.</param>
+        /// <param name="provider">An object that supplies culture-specific format information about s.</param>
+        /// <param name="style">A bitwise combination of enumeration values that indicates the permitted format of s. A typical value to specify is None.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by provider and styles.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s, IFormatProvider? provider, DateTimeStyles style = DateTimeStyles.None)
+        {
+            if ((style & ~DateTimeStyles.AllowWhiteSpaces) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidDateStyles, nameof(style));
+            }
+
+            DateTimeResult result = default;       // The buffer to store the parsing result.
```

Comment:
Please remove obvious comment.
```suggestion
            DateTimeResult result = default;
```